railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -115,6 +117,17 @@ def extract_environment_from_path(path)
           available_environments.find { |env| path.include? env } if path.end_with?(".yml.enc")
         end
 
+        # Tries to check for mismatch between the provided `environment` class_option and the
+        # Rails::Command.environment. When no class_option for environment is provided, by default,
+        # it works with the development environment. So, ideally warning the user on their performing
+        # action could avoid unnecessary confusions. This warning happens only in the non-development
+        # environment.
```

Comment:
I think this method is simple enough that this comment is almost unnecessary, since it's private only Rails devs will see it.

If we do want to add a comment, I would try to rephrase it to be more concise, for example:

> Check for a mismatch between Rails::Command.environment and option set by --environment flag. This warning should only occur in non-development environments to avoid confusing the user when trying to perform this action against different environment modes.

Reply:
Updated the comments.